device/dell/x86_64-dellemc_z9432f_c3758-r0/plugins/fanutil.py

```diff
@@ -20,26 +20,30 @@ class FanUtil(FanBase):
     NUM_FANS_PERTRAY = 2
     FANTRAY_NUM_START_IDX = 1
     FRU_FAN_START_IDX = 1
-    IPMI_FAN_PRESENCE = "ipmitool sensor get FAN{0}_prsnt"
-    IPMI_FAN_FRONT_SPEED = "ipmitool sdr get Fan{0}_Front_rpm"
-    IPMI_FAN_REAR_SPEED = "ipmitool sdr get Fan{0}_Rear_rpm"
-    IPMI_FRU_DATA = "ipmitool fru print {0}"
+    IPMI_FAN_PRESENCE = ["ipmitool", "sensor", "get", "FAN"]
+    IPMI_FAN_FRONT_SPEED = ["ipmitool", "sdr", "get", "Fan"]
+    IPMI_FAN_REAR_SPEED = ["ipmitool", "sdr", "get", "Fan"]
+    IPMI_FRU_DATA = ["ipmitool", "fru", "print", ""]
 
     def __init__(self, log_level=logging.DEBUG):
         FanBase.__init__(self)
         self.num_fans = (self.FANTRAY_NUM_ON_MAIN_BOARD*self.NUM_FANS_PERTRAY)
 
     def get_fan_status(self,fan_id):
         try:
-           ret_status, ipmi_cmd_ret = commands.getstatusoutput(self.IPMI_FAN_PRESENCE.format(fan_id))
+            self.IPMI_FAN_PRESENCE[3] += str(fan_id) + '_prsnt'
+            p = subprocess.run(self.IPMI_FAN_PRESENCE, capture_output=True, universal_newlines=True)
+            ret_status, ipmi_cmd_ret = p.returncode, p.stdout
```

Comment:
>             [](http://example.com/codeflow?start=0&length=12)

The indentation is different from next line. Will it work?

Reply:
Fixed indentation error